frontend/src/metabase/lib/expressions/config.js

```diff
@@ -363,12 +363,12 @@ export const MBQL_CLAUSES = {
   },
   "date-add": {
     displayName: `dateAdd`,
-    type: "number",
+    type: "expression",
```

Comment:
Follow the same pattern as for `case` and `coalesce`:
- Type these as `expression`
- Add special handling in `infer` and use the type of the first argument